### PR TITLE
Brokers must have at least one service, and services must have at least one plan

### PR DESCRIPTION
**Pull request by mattmcneeney**
https://github.com/openservicebrokerapi/servicebroker/issues/206

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -111,7 +111,7 @@ A web-friendly display name is camel-cased with spaces and punctuation supported
 
 | Response field | Type | Description |
 | --- | --- | --- |
-| services* | array-of-service-objects | Schema of service objects defined below. |
+| services* | array-of-service-objects | Schema of service objects defined below. MUST contain at least one service. |
```
Comment (angarg12) [inline spec.md]: Small wording, should we say 'MUST be non-empty' or the like instead?

Comment (avade) [inline spec.md]: I think as the type is a list of services, specifying at least one service makes sense.

**Comment by angarg12** [comment]
LGTM

**Comment by duglin** [comment]
hmm, if we require non-empty service lists what does the broker return if it has none? Some 4xx/5xx error?  Feel odd to return an error when its not really an error, its possible a broker could return a non-zero list for other platforms, just not this current platform.  Perhaps empty service list really is ok.

**Comment by mattmcneeney** [comment]
That's true. I hadn't thought of a use case for a broker returning no services, but it makes sense if it was only supporting a subset of the platforms that could call it. I'll update this PR.

**Comment by mattmcneeney** [comment]
However you could make the same argument for plans, right? Is this PR redundant?

**Comment by duglin** [comment]
I think plans are different because a platform can't do anything with a service that doesn't have any plans, so returning the service at that point is pointless.

So perhaps this PR should just be:
1 - a service w/o any plans MUST be excluded
2 - a service list MAY be empty.

**Comment by mattmcneeney** [comment]
I changed this so that services `MAY be empty` and plans `MUST contain at least one plan`. I think that covers those points?

**Comment by duglin** [comment]
Yup it implies it, so unless someone wants it stated explicitly....
LGTM

**Comment by vaikas-google** [comment]
LGTM

**Comment by avade** [comment]
lgtm

**Comment by angarg12** [comment]
LGTM

**Pull request merged**